### PR TITLE
Add batch unsubscribing functionality for gitea

### PR DESCRIPTION
**Pull request by RealSaltyFish**
This PR adds a `unsubscribe-from-repos` command that helps TAs easily unsubscribe from student repos to avoid being email flooded by gitea.

Note: line break issue is not fixed. It is worked around for now, and needs a force push to be removed from the commit history. This PR might also break things since there were a lot of merge conflicts. Please review carefully.

### CODE REVIEW

File: joint_teapot/workers/gitea.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+<<<<<<< HEAD
```
Comment (BoYanZh) [inline joint_teapot/workers/gitea.py]: remove

File: .vimspector.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (BoYanZh) [inline .vimspector.json:1]: remove

**Review by BoYanZh — CHANGES_REQUESTED** [review]

File: joint_teapot/workers/gitea.py (hunk 1 above)
Comment (CZ-He) [reply joint_teapot/workers/gitea.py]: I did not push this... Appears that you forgot to remove it in a merge. Anyway I fixed it for you.

Comment (RealSaltyFish) [reply joint_teapot/workers/gitea.py]: I did not push this... Appears that you forgot to remove it in a merge. Anyway I fixed it for you.

Comment (BoYanZh) [reply joint_teapot/workers/gitea.py]: https://github.com/RealSaltyFish/Joint-Teapot/commit/667b9cd53baaeab39292687f90dc400b288437d1

File: joint_teapot/workers/gitea.py [hunk 3]
```diff
@@ -401,8 +396,29 @@ def get_all_teams(
...
+        regex = re.compile(pattern)
+        subscriptions = [
+            sub
+            for sub in self.user_api.user_current_list_subscriptions()
+            if sub.owner.login == self.org_name
+            and re.search(regex, sub.name) is not None
+        ]
+        if len(subscriptions) == 0:
+            logger.warning(f"No subscribed repo matches the pattern {pattern}")
+        else:
```
Comment (BoYanZh) [inline joint_teapot/workers/gitea.py:413]: just `return`

File: joint_teapot/workers/gitea.py [hunk 4]
```diff
@@ -401,8 +396,29 @@ def get_all_teams(
...
+            logger.info(
+                f"{len(subscriptions)} subscriptions match the pattern {pattern}: {subscriptions[0].name} ... {subscriptions[-1].name}"
```
Comment (BoYanZh) [inline joint_teapot/workers/gitea.py:415]: better log them all

File: joint_teapot/workers/gitea.py [hunk 5]
```diff
@@ -401,8 +396,29 @@ def get_all_teams(
...
+        return res
+
+
+    def unsubscribe_from_repos(self, pattern: str) -> None:
+        regex = re.compile(pattern)
+        subscriptions = [
+            sub
+            for sub in self.user_api.user_current_list_subscriptions()
+            if sub.owner.login == self.org_name
+            and re.search(regex, sub.name) is not None
```
Comment (BoYanZh) [inline joint_teapot/workers/gitea.py:409]: `re.search(pattern, sub.name)`

**Review by BoYanZh — CHANGES_REQUESTED** [review]

File: joint_teapot/workers/gitea.py (hunk 1 above)
Comment (RealSaltyFish) [reply joint_teapot/workers/gitea.py]: resolved

File: .vimspector.json (hunk 2 above)
Comment (RealSaltyFish) [reply .vimspector.json]: resolved

File: joint_teapot/workers/gitea.py (hunk 5 above)
Comment (RealSaltyFish) [reply joint_teapot/workers/gitea.py:409]: done

File: joint_teapot/workers/gitea.py (hunk 4 above)
Comment (RealSaltyFish) [reply joint_teapot/workers/gitea.py:415]: done

File: joint_teapot/workers/gitea.py (hunk 3 above)
Comment (RealSaltyFish) [reply joint_teapot/workers/gitea.py:411]: Can you remind me again where this was? I did a force push and the line number changed.

Comment (RealSaltyFish) [reply joint_teapot/workers/gitea.py]: done

**Pull request merged**